File: VB/ReportForQueryExample/Controllers/HomeController.vb
Public Class HomeController
	Inherits Controller

	Public Function Index() As ActionResult
		Return View()
	End Function

	Public Function Viewer() As ActionResult
		If Request.QueryString("TableName") Is Nothing Then
			Response.Redirect("Viewer?TableName=Customers")
		Else
			Dim tableName As String = Request.QueryString("TableName")
			Dim report As New TestReport(tableName)
			Return View(report)
		End If
		Return View(New DevExpress.XtraReports.UI.XtraReport())
	End Function
End Class

File: CS/ReportForQueryExample/Controllers/HomeController.cs
using System;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace ReportForQueryExample.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult Viewer() {
            if (Request.QueryString["TableName"] == null)
                Response.Redirect("Viewer?TableName=Customers"); 
            else
            {
                string tableName = Request.QueryString["TableName"];
                TestReport report = new TestReport(tableName);
                return View(report);
            }
            return View(new DevExpress.XtraReports.UI.XtraReport());
        }
    }
}
